src/manifolds/ProbabilitySimplex.jl

```diff
@@ -70,38 +70,29 @@ active_traits(f, ::ProbabilitySimplex, args...) = merge_traits(IsEmbeddedManifol
     change_representer(M::ProbabilitySimplex, ::EuclideanMetric, p, X)
 
 Given a tangent vector with respect to the metric from the embedding, the [`EuclideanMetric`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/manifolds.html#ManifoldsBase.EuclideanMetric),
-the representer of a linear functional on the tangent space is adapted as ``Z = p .* X``, since
-this “compensates” for the divsion by ``p`` in the Riemannian metric on the [`ProbabilitySimplex`](@ref).
+the representer of a linear functional on the tangent space is adapted as ``Z = p .* X .- p .* dot(p, X)``.
+The first part “compensates” for the divsion by ``p`` in the Riemannian metric on the [`ProbabilitySimplex`](@ref)
+and the second part performs appropriate projection to keep the vector tangent.
 
-To be precise for any ``Y ∈ T_pΔ^n`` we are looking for ``Z ∈ T_pΔ^n`` such that
-
-```math
-    ⟨X,Y⟩ = X^\mathrm{T}Y = \sum_{i=1}^{n+1}\frac{Z_iY_i}{p_i} = g_p(Z,Y)
-```
-
-and hence ``Z_i = X_ip_i, i=1,…,n+1``.
+For details see Proposition 2.3 in [^ÅströmPetraSchmitzerSchnörr2017].
 """
 change_representer(::ProbabilitySimplex, ::EuclideanMetric, ::Any, ::Any)
 
 function change_representer!(::ProbabilitySimplex, Y, ::EuclideanMetric, p, X)
-    return Y .= p .* X
+    return Y .= p .* X .- p .* dot(p, X)
 end
 
 @doc raw"""
     change_metric(M::ProbabilitySimplex, ::EuclideanMetric, p, X)
 
 To change the metric, we are looking for a function ``c\colon T_pΔ^n \to T_pΔ^n`` such that for all ``X,Y ∈ T_pΔ^n``
-
-```math
-    ⟨X,Y⟩ = X^\mathrm{T}Y = \sum_{i=1}^{n+1}\frac{c(X)_ic(Y)_i}{p_i} = g_p(X,Y)
-```
-
-and hence ``C(X)_i = X_i\sqrt{p_i}, i=1,…,n+1``.
+This can be achieved by rewriting representer change in matrix form as `(Diagonal(p) - p * p') * X`
+and taking square root of the matrix
 """
 change_metric(::ProbabilitySimplex, ::EuclideanMetric, ::Any, ::Any)
 
 function change_metric!(::ProbabilitySimplex, Y, ::EuclideanMetric, p, X)
-    return Y .= sqrt.(p) .* X
+    return Y .= sqrt(Diagonal(p) - p * p') * X
```

Comment:
Here it would be nice to know whether this really holds. That the metric is preserved is clear to me (I think); since `X` (and `Y`) are tangents the last three summands of X'R'RY (R is the matrix we define here, and you can rotate ) should vanish. Whether the results here are tangent I do not see that fast.

Reply:
A rough sketch: `w = [1, 1, ..., 1]` is an eigenvector of `R=Diagonal(p) - p * p'` with eigenvalue 0 because it's a projection onto the tangent space. So `w` is also an eigenvector of `sqrt(R)`, with the same eigenvalue. So `sqrt(R)` cannot add a component in the direction `w` to a vector it acts on.

Reply:
I think this should work. To me this looks like a projection matrix, so it is symmetric. I am not so sure its positive semidefinite – but then the square root is unique, see https://en.wikipedia.org/wiki/Square_root_of_a_matrix#Matrices_with_distinct_eigenvalues – further you can use the eigenvalue decomposition to compute this unique square root – and then the square roots of the eigenvalues of A are the eigenvalues of sqrt(A). This would keep your zero as an eigenvalue with the same eigenvector.

So I think (positive semidefiniteness left) this holds true.